src/lib/u2/browser.js

```diff
@@ -0,0 +1,550 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'DAOController',
+  extends: 'foam.u2.View',
+  // Expects a DAO as its data. Calls listFactory to construct the list, passing
+  // it rowFactory.
+  // Clicking a row creates a DAOUpdateController to edit the item (lists should
+  // publish rowClick, like DAOList does).
+  // Also displays a floating action button to create new items, which uses
+  // DAOCreateController.
+  requires: [
+    'foam.u2.DAOCreateController',
+    'foam.u2.DAOList',
+    'foam.u2.DAOUpdateController'
+  ],
+
+  imports: [
+    'stack'
+  ],
+
+  properties: [
+    {
+      name: 'of',
+      expression: function(data) {
+        return data.of;
+      }
+    },
+    {
+      name: 'data'
+    },
+    {
+      class: 'foam.u2.ViewFactory',
+      name: 'listFactory',
+      value: 'foam.u2.DAOList'
+    },
+    {
+      class: 'foam.u2.ViewFactory',
+      name: 'rowFactory',
+      value: 'foam.u2.CitationView'
+    }
+  ],
+
+  listeners: [
+    {
+      name: 'rowClick',
+      code: function(sub, _, obj) {
+        this.stack.pushAfter(this, this.DAOUpdateController.create({
+          of: this.of,
+          data: obj,
+          dao: this.data
+        }));
+      }
+    }
+  ],
+
+  actions: [
+    {
+      name: 'newItem',
+      label: 'New',
+      code: function() {
+        this.stack.pushAfter(this, this.DAOCreateController.create({
+          of: this.of,
+          dao: this.data
+        }));
+      }
+    }
+  ],
+
+  methods: [
+    function initE() {
+      var list = this.listFactory$f({
+        rowFactory: this.rowFactory,
+        data: this.data
+      }, this.__subContext__);
+      list.rowClick.sub(this.rowClick);
+      var X = this.__subContext__.createSubContext({ data: this });
+      this.cssClass(this.myCls()).add(this.cls_.NEW_ITEM.toE(X)).add(list);
```

Comment:
U2 now supports .startContext() and .endContext().

.startContext({ data: this }).add(this.cls_.NEW_ITEM).endContext();

Even better might be to use .start(this.cls_.NEW_ITEM, { data: this }), but it looks like support for that hasn't been pushed yet.

Maybe add a TODO to update this to modern U2 style when its ready.